.github/workflows/comment_on_pr.yaml

```diff
@@ -50,12 +50,13 @@ jobs:
           echo "Head SHA: $HEAD_SHA"
           echo "HEAD_SHA=$HEAD_SHA" >> "$GITHUB_ENV"
 
-      - name: Delete old comments
-        uses: izhangzhihao/delete-comment@master
+      - name: Find Comment
+        uses: peter-evans/find-comment@v3
+        id: fc
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          delete_user_name: github-actions[bot]
-          issue_number: ${{ env.PR_NUMBER }}
+          issue-number: ${{ env.PR_NUMBER }}
+          comment-author: 'github-actions[bot]'
+          body-includes: "Build Successful!"
```

Comment:
This won't find the comment if the build failed, no?

Reply:
Yes? But that seems to be the behavior already? It's probably worth adding a check and saying it failed anyway.

Reply:
ok so i just confirmed it, because of the conditional statement on line 13, it will never run on a failed build